server/src/interfaces/notification.interface.ts

```diff
@@ -0,0 +1,51 @@
+export const INotificationRepository = 'INotificationRepository';
+
+export type SendEmailOptions = {
+  from: string;
+  to: string;
+  replyTo?: string;
+  subject: string;
+  html: string;
+  text: string;
+  smtp: SmtpOptions;
+};
+
+export type SmtpOptions = {
+  host: string;
+  port?: number;
+  username?: string;
+  password?: string;
+  ignoreCert?: boolean;
+};
+
+export enum EmailTemplate {
+  WELCOME = 'welcome',
+  RESET_PASSWORD = 'reset-password',
+}
+
+export interface WelcomeEmailProps {
+  baseUrl: string;
+  displayName: string;
+  username: string;
+  password?: string;
+}
+
+export interface ResetPasswordEmailProps {
+  user: string;
+  email: string;
+}
+
+export type EmailRenderRequest =
+  | { template: EmailTemplate.WELCOME; data: WelcomeEmailProps }
+  | { template: EmailTemplate.RESET_PASSWORD; data: ResetPasswordEmailProps };
```

Comment:
We have a reset password email template here but I don't see it used anywhere else?

Reply:
Not yet. I added this to make sure the workflow and types worked with more than one, but I can remove it now.